### DISCUSSION TITLE
Remove @Deprecated(forRemoval = true) from org.bukkit.event.entity.EntityKnockbackEvent

### DISCUSSION THREAD

**Discussion by Ynverxe** [PaperMC/Paper#10604] [Ideas]
### Is your feature request related to a problem?

Actually Paper API has two events that are called when an entity is pushed or receives knockback from another entity.
This is good because EntityKnockbackByEntity of spigot doesn't allow non living entities. So I think that may be the reason why this event is marked as deprecated, it's just my idea.

EntityKnockbackEvent is very usefull to handle explosion that doesn't have a source, like an explosion by a Respawn Anchor or artificial explosions created by World#createExplosion.

The deprecation on EntityKnockbackEvent was introduced in this commit: https://github.com/PaperMC/Paper/pull/10242/commits/f1721555c9fa13ff0ac0998dda02141e0db143c3

### Describe the solution you'd like.

Just remove the @Deprecated annotation.

### Describe alternatives you've considered.

No other.

**Comment by electronicboy** [comment]
The event is deprecated because it duplicates an existing paper event, if the paper events are missing a case, an issue should be filed for that so that can be mitigated, there's no point having multiple events for the same thing

**Comment by Ynverxe** [comment]
> The event is deprecated because it duplicates an existing paper event, if the paper events are missing a case, an issue should be filed for that so that can be mitigated, there's no point having multiple events for the same thing

If so, then it occurs to me to create an event that covers explosions with and without a source that includes a `@Nullable getSource()` method.

EDIT: But I suppose this would cause a deprecation of the already existing paper event

**Comment by Ynverxe** [comment]
It might be useful to create an event that is invoked during an explosion, regardless of where it came from. Mainly, because although I know that there are currently events that are called on similar occasions, however, these events such as `EntityExplodeEvent`, `BlockExplodeEvent` and `TNTPrimeEvent` do not allow modifying explosion values, such as its radius or the push that the entities receive. This does not conflict with what the friend @electronicboy commented. I would appreciate the feedback ^^

**Comment by Machine-Maker** [comment]
We added our own event, called in the same places with same reasons in https://github.com/PaperMC/Paper/pull/10831.
